### ISSUE TITLE
sample_smc

### ISSUE THREAD

**Issue by myravian** [pymc-devs/pymc-experimental#319]
Hi there, 

I was testing pymc_experimental/inference/smc/sampling.py and noticed the following issues:
- the inference doesn't seem to like pm.Dirichlet, with a shape error at tmp = logp_fn(*[p.squeeze() for p in particles])[0]
- arviz_from_particles doesn't seem to like RVs with shape=(1,) 
- the conversion from inferencedata to netCDF fails because the integrations is neither int nor np.array
- the inferencedata doesn't have the marginal likelihood, do you think it will be implemented in the future or it's just not possible?

Thanks a lot for the SMC blackjax implementation, it's very useful!

Cheers,
VIan

PS: here's some code that produces the error

`real_a = 0.2
real_b = 2
x = np.linspace(1, 100)
y = real_a * x + real_b + np.random.normal(0, 2, len(x))

with pm.Model() as model:
    a = pm.Normal("a", mu=10, sigma=10)
    b = pm.Normal("b", mu=10, sigma=10)
    # either of the following lines produces an error
    # c = pm.Normal("c", mu=10, sigma=10, shape=(1,))
    # d = pm.Dirichlet("d", [1, 1])

    trace = sample_smc(
        n_particles=1000,
        kernel="HMC",
        inner_kernel_params={
            "step_size": 0.01, 
            "integration_steps": 20,
        },
        iterations_to_diagnose=10,
        target_essn=0.5,
        num_mcmc_steps=10,
    )
`

**Comment by ciguaran** [comment]
Hi, I can tackle this could someone assign the issue to me?

**Comment by ciguaran** [comment]
@ricardoV94 I would like your input here, since this is related to Pytensor. What should be the expected parameters shape when using get_jaxified_graph https://github.com/pymc-devs/pymc-experimental/blob/main/pymc_experimental/inference/smc/sampling.py#L384C15-L384C33 thanks!

**Comment by ciguaran** [comment]
@myravian could you try your example running it from this branch? I may have a fix https://github.com/ciguaran/pymc-experimental/tree/ciguaran_fix_smc_bj . Also super interested to know what are you using SMC for, it would be great if it would become an example notebook on how to use it!. let me know.

**Comment by ciguaran** [comment]
Could you share a full python file that reproduces the error? I've run the example you posted at the very beginning and it does work for me 🤔 .

**Comment by myravian** [comment]
Here would be the script:
`import pymc as pm

from sampling_smc_ciguaran import sample_smc_blackjax as sample_smc

with pm.Model() as model:
    c = pm.Normal("c", mu=10, sigma=10, shape=(1,))
    d = pm.Dirichlet("d", [1, 1])

    trace = sample_smc(
        n_particles=1000,
        kernel="HMC",
        inner_kernel_params={
            "step_size": 0.01,  # small values better
            "integration_steps": 20,
        },
        iterations_to_diagnose=10,
        target_essn=0.5,
        num_mcmc_steps=10,
    )
`
Maybe it has to do with the blackjax/jax versions (1.1.0/0.4.21 in my system)

**Comment by ciguaran** [comment]
Hi! so I was able to run the example you just shared via installing pymc-experimental from the branch.

`pip install git+https://github.com/ciguaran/pymc-experimental@ciguaran_fix_smc_bj` 

is it possible that you are still using pymc-experimental from master?

**Comment by myravian** [comment]
You're right, I was not using the proper versions, just tested it and it seems to work fine, thanks for the modifications!
